adapter/internal/synchronizer/apis_fetcher.go

```diff
@@ -160,34 +161,48 @@ func MergeDeployedRevisionList(deployedRevisionList []*notifier.DeployedAPIRevis
 // given API ID and a list of environments that API has been deployed to.
 // updatedAPIID is the corresponding ID of the API in the form of an UUID
 // updatedEnvs contains the list of environments the API deployed to.
-func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string) {
+func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string, envToDpMap msg.GatewayEnvToDataPlaneIDMap) {
 	// Read configurations and derive the eventHub details
 	conf, errReadConfig := config.ReadConfigs()
 	if errReadConfig != nil {
 		// This has to be error. For debugging purpose info
 		logger.LoggerSync.Errorf("Error reading configs: %v", errReadConfig)
 	}
 	configuredEnvs := conf.ControlPlane.EnvironmentLabels
-	//finalEnvs contains the actual envrionments that the adapter should update
+	//finalEnvs contains the actual environments that the adapter should update
 	var finalEnvs []string
-	if len(configuredEnvs) > 0 {
-		// If the configuration file contains environment list, then check if then check if the
-		// affected environments are present in the provided configs. If so, add that environment
-		// to the finalEnvs slice
+
+	if conf.ControlPlane.DynamicEnvironments.Enabled {
 		for _, updatedEnv := range updatedEnvs {
```

Comment:
instead iterating through updatedEnvs, we can iterate through `envToDpMap`

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/3212/commits/7d325deba6d2def0dcc0baadc01b2db643061ee2